### PR TITLE
Tianqihu/azure sdk java fix atts

### PR DESCRIPTION
**Pull request by tianqihu7**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by tianqihu7** [comment]
@microsoft-github-policy-service agree company="Microsoft"

File: sdk/attestation/azure-security-attestation/src/main/java/com/azure/security/attestation/AttestationAsyncClient.java [hunk 1]
```diff
@@ -534,71 +535,43 @@ Mono<AttestationResponse<AttestationResult>> attestSgxEnclaveWithResponse(Attest
...
-     * @throws HttpResponseException thrown if the request is rejected by server.
-     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
      * @return attestation response for Trusted Platform Module (TPM) attestation.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws HttpResponseException    thrown if the request is rejected by server.
+     * @throws RuntimeException         all other wrapped checked exceptions if the request fails to be sent.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<String> attestTpm(String request) {
+    public Mono<TpmAttestationResult> attestTpm(BinaryData request) {
```
Comment (LarryOsterman) [inline sdk/attestation/azure-security-attestation/src/main/java/com/azure/security/attestation/AttestationAsyncClient.java:566]: If the is `BinaryData`, the output should also be `BinaryData`. If the Output is `TpmAttestationResult`, the input should be `TpmAttestationRequest`.

Be consistent with both of these.

File: sdk/attestation/azure-security-attestation/src/main/java/com/azure/security/attestation/AttestationAsyncClient.java [hunk 2]
```diff
@@ -534,71 +535,43 @@ Mono<AttestationResponse<AttestationResult>> attestSgxEnclaveWithResponse(Attest
...
 
-    /** Performs TPM attestation.
-     *
-     * Processes attestation evidence from a VBS enclave, producing an attestation result.
-     * <p>The TPM attestation protocol is defined <a href='https://docs.microsoft.com/azure/attestation/virtualization-based-security-protocol'>here.</a></p>
-     * <p>Unlike OpenEnclave reports and SGX enclave quotes, TPM attestation is implemented using JSON encoded
-     * strings. </p><p>The client formats a string serialized JSON request to the service, which responds with a
-     * JSON response. The serialized JSON object exchange continues until the service responds with a JSON string
-     * with a property named {@code "report"}, whose value will be an attestation result token.</p>
-     * <p><strong>Perform the first leg of a TPM attestation operation</strong></p>
```
Comment (LarryOsterman) [inline sdk/attestation/azure-security-attestation/src/main/java/com/azure/security/attestation/AttestationAsyncClient.java:545]: Note that the conventions for Java require that all APIs have snippets showing the use of the API, I believe you need to update the snippet here.

**Pull request closed without merging**